<a href="https://colab.research.google.com/github/jackthapakron/tool/blob/main/Tool_Homework1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
# 1 Question

import numpy as np

def count_runs(sequence):
    run_lengths = []
    current_run = 0

    for num in sequence:
        if num == 1:
            current_run += 1
        elif current_run > 0:
            run_lengths.append(current_run)
            current_run = 0

    if current_run > 0:
        run_lengths.append(current_run)

    run_counts = {}
    for length in run_lengths:
        if length in run_counts:
            run_counts[length] += 1
        else:
            run_counts[length] = 1

    return run_counts


result1 = count_runs([0, 1, 0, 1, 1, 0, 0, 0, 0, 1])
result2 = count_runs(np.random.randint(0, 2, 1000000))

print(result1)
print(result2)


{1: 2, 2: 1}
{2: 62558, 1: 125134, 3: 31386, 4: 15572, 6: 3938, 5: 7768, 7: 1943, 10: 227, 8: 944, 9: 481, 11: 140, 15: 7, 14: 12, 13: 36, 12: 65, 18: 3, 19: 1, 16: 4, 21: 1, 17: 1}


In [ ]:
result = count_runs([0, 1, 1, 1, 1, 0, 0, 1, 0, 1, 0, 0, 0,1,1,1,1,1])
print(result)


{4: 1, 1: 2, 5: 1}


In [ ]:
#2 Question

import numpy as np

def run_prob(sequence, k):
    run_counts = count_runs(sequence)

    for length, count in run_counts.items():
        if length >= k:
            return True

    return False


experiments = 100000
success_count = sum(run_prob(np.random.choice([0, 1], size=100, p=[0.5, 0.5]), k=5) for _ in range(experiments))
result1 = success_count / experiments

success_count = sum(run_prob(np.random.choice([0, 1], size=100, p=[0.7, 0.3]), k=7) for _ in range(experiments))
result2 = success_count / experiments

print(result1)
print(result2)

0.81151
0.01512


**2 Question**


*   The probability of observing at least one run of length 5 or more when n=100 and p=0.5 is approximately 0.81151.
*   The probability of observing at least one run of length 7 or more when p=0.7 is approximately 0.01512.




Comparing these probabilities, it appears that the likelihood of observing a run of length 5 or more is significantly higher than the likelihood of observing a run of length 7 or more.
